### ISSUE TITLE
Correctly distinguish dict/sets

### ISSUE THREAD

**Issue by isidentical** [davidhalter/parso#126]
I was checking the `check_assignment` function and then noticed it isn't able to detect dict/sets fully (this is a really rare case, I don't even know if it worths fixing)
```
>>> get_errors("{a:b} = 1")[0].message
'SyntaxError: cannot assign to dict display'
>>> get_errors("{**x} = 1")[0].message
'SyntaxError: cannot assign to set display'
```

**Comment by davidhalter** [comment]
Yeah, we should probably fix that. Good catch!

**Comment by isidentical** [comment]
> Yeah, we should probably fix that. Good catch!
I'm working on a [get_expr_name](https://github.com/python/cpython/blob/800a35c623bbcdb5793c7d7a4974524286311479/Parser/pegen/pegen.c#L161-L233) alternative (for 3.9+ error messages). Will fix it on the way

**Comment by davidhalter** [comment]
Great!

**Comment by isidentical** [comment]
```
>>> get_errors("f'adsdsa' = 1")
[]
```

**Comment by isidentical** [comment]
Fixed with #127
